controllers/nodefeaturediscovery_finalizers.go

```diff
@@ -0,0 +1,411 @@
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteClusterRoleBinding(ctx, instance.ObjectMeta.Namespace, nfdMasterApp)
+		if err != nil {
+			return false, interpretError(err, "ClusterRoleBinding")
+		}
+		klog.Info("ClusterRoleBinding", nfdMasterApp, "resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Attempt to delete the Worker ServiceAccount
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteServiceAccount(ctx, instance.ObjectMeta.Namespace, nfdWorkerApp)
+		if err != nil {
+			return false, interpretError(err, "worker ServiceAccount")
+		}
+		klog.Info("Worker ServiceAccount resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Attempt to delete the Master ServiceAccount
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteServiceAccount(ctx, instance.ObjectMeta.Namespace, nfdMasterApp)
+		if err != nil {
+			return false, interpretError(err, "master ServiceAccount")
+		}
+		klog.Info("Master ServiceAccount resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Attempt to delete the Worker config map
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteConfigMap(ctx, instance.ObjectMeta.Namespace, nfdWorkerApp)
+		if err != nil {
+			return false, interpretError(err, "nfd-worker config map")
+		}
+		klog.Info("nfd-worker config map resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// If NFD-Topology-Updater was requested
+	if instance.Spec.TopologyUpdater {
+		// Attempt to delete Topology DaemonSet
+		err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+			err = r.deleteDaemonSet(ctx, instance.ObjectMeta.Namespace, nfdTopologyUpdaterApp)
+			if err != nil {
+				return false, interpretError(err, "Topology DaemonSet")
+			}
+			klog.Info("nfd-Topology-Updater DaemonSet resource has been deleted.")
+			return true, nil
+		})
+		if err != nil {
+			return err
+		}
+		// Attempt to delete the ClusterRole
+		err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+			err = r.deleteClusterRole(ctx, instance.ObjectMeta.Namespace, nfdTopologyUpdaterApp)
+			if err != nil {
+				return false, interpretError(err, "nfd-Topology-Updater ClusterRole")
+			}
+			klog.Info("nfd-Topology-Updater ClusterRole resource has been deleted.")
+			return true, nil
+		})
+		if err != nil {
+			return err
+		}
+		// Attempt to delete the ClusterRoleBinding
+		err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+			err = r.deleteClusterRoleBinding(ctx, instance.ObjectMeta.Namespace, nfdTopologyUpdaterApp)
+			if err != nil {
+				return false, interpretError(err, "ClusterRoleBinding")
+			}
+			klog.Info("ClusterRoleBinding", nfdTopologyUpdaterApp, "resource has been deleted.")
```

Comment:
spaces around words?

```suggestion
			klog.Info("ClusterRoleBinding ", nfdTopologyUpdaterApp, " resource has been deleted.")
```